app/src/org/commcare/android/resource/installers/ProfileAndroidInstaller.java

```diff
@@ -119,6 +130,21 @@ public boolean install(Resource r, ResourceLocation location, Reference ref,
         return false;
     }
 
+    private void storeReleasedTime(AndroidCommCarePlatform platform, Reference ref) throws ParseException {
+        long releasedOnTime = -1;
+
+        if (ref instanceof ReleasedOnTimeSupportedReference) {
+            releasedOnTime = ((ReleasedOnTimeSupportedReference)ref).getReleasedOnTime();
+        }
+
+        // If we don't have the release time, guess it as current time
+        if (releasedOnTime == -1) {
```

Comment:
I think this fallback isn't exactly right.

As-is, this change will immediately force every app to sync before update, which I don't think is quite the intent. 

I think this fallback is good if there is some other setting which controls whether the "Sync before Update" behavior triggers (like an app setting), which implies that the fallback for having a released on date should be syncing every time, but otherwise I think we should (until someone from the requirements team directs otherwise) not change the default behavior.

Reply:
@ctsims  We do have [this custom setting](https://github.com/dimagi/commcare-android/blob/appUpdateAndUserSync/app/src/org/commcare/preferences/HiddenPreferences.java#L363) `pre-update-sync-needed` which is needed currently to turn on the "Sync before update" workflow apart from the released on date.